### PR TITLE
[shopsys] remake of model documentation

### PR DESCRIPTION
**Pull request by MattCzerner**
| Q             | A
| ------------- | ---
|Description, reason for the PR| After many modifications of a model, there was a need to review the current documentation about model and to add some rules and information about what is, and what is not model.
|New feature| No
|[BC breaks](https://github.com/shopsys/shopsys/blob/master/docs/contributing/backward-compatibility-promise.md)| No
|Fixes issues| 
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

I didnt decide to move classes that does not belong to the model because of unnecessary BC break, I will create a USies for that, so we can do it later.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -44,6 +43,9 @@
 * [Creating a Multidomain Design](./cookbook/creating-a-multidomain-design.md)
 * [Working with Multiple Cron Instances](./cookbook/working-with-multiple-cron-instances.md)
 
+## Model
+* [Basics About Model Architecture](./model/introduction-to-model-architecture.md)
```
Comment (boris-brtan) [inline docs/index.md]: please change all `basics-about-model-architecture` into `introduction-to-model-architecture`

File: docs/cookbook/basic-data-import.md [hunk 2]
```diff
@@ -168,8 +168,8 @@ private function importExternalProductsData(array $externalProductsData)
 ```
 
 *Note: We need to know whether the product with given `$extId` exists.
-For that purpose, we will use a descendant of [`ProductFacade`](../../packages/framework/src/Model/Product/ProductFacade.php) ([more about facades](../introduction/basics-about-model-architecture.md#facade))
-which will use a descendant of [`ProductRepository`](../../packages/framework/src/Model/Product/ProductRepository.php) ([more about repositories](../introduction/basics-about-model-architecture.md#repository))
+For that purpose, we will use a descendant of [`ProductFacade`](../../packages/framework/src/Model/Product/ProductFacade.php) ([more about facades](../model/basics-about-model-architecture.md#facade))
```
Comment (boris-brtan) [inline docs/cookbook/basic-data-import.md]: file does not exist

File: docs/introduction/basics-about-package-architecture.md [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 ## Introduction
 This article describes the architecture of Shopsys Framework from the viewpoint of packages
-(in a contrast to application layers, about which you can read in the [separate article](./basics-about-model-architecture.md)).
+(in a contrast to application layers, about which you can read in the [separate article](../model/basics-about-model-architecture.md)).
```
Comment (boris-brtan) [inline docs/introduction/basics-about-package-architecture.md]: file not found

File: docs/introduction/components.md [hunk 4]
```diff
@@ -1,7 +1,7 @@
 # Components
 
 ## Definition
-Components are classes usable for any type of applications, not only e-commerce projects(unlike [Application model](./basics-about-model-architecture.md)),
+Components are classes usable for any type of applications, not only e-commerce projects(unlike [Application model](../model/basics-about-model-architecture.md)),
```
Comment (boris-brtan) [inline docs/introduction/components.md]: not found

File: docs/introduction/entities.md [hunk 5]
```diff
@@ -1,7 +1,7 @@
 # Entities
 
 This article describes how we work with entities and our specialities.
-1. Entity is a class encapsulating data and you can read more what is an entity in the [model architecture article](basics-about-model-architecture.md).
+1. Entity is a class encapsulating data and you can read more what is an entity in the [model architecture article](../model/basics-about-model-architecture.md).
```
Comment (boris-brtan) [inline docs/introduction/entities.md]: not found

File: docs/introduction/faq-and-common-issues.md [hunk 6]
```diff
@@ -93,7 +93,7 @@ Yes, there is some prepared configuration for Shopsys Framework cron commands in
 Do not forget to set up a cron on your server to execute [`php phing cron`](/docs/introduction/console-commands-for-application-management-phing-targets.md#cron) every 5 minutes.
 
 ## Why are you using entity data instead of entities for Symfony forms?
-We are using [entity data objects](entities.md#entity-data) instead of [entities](basics-about-model-architecture.md#entity)
+We are using [entity data objects](entities.md#entity-data) instead of [entities](../model/basics-about-model-architecture.md#entity)
```
Comment (boris-brtan) [inline docs/introduction/faq-and-common-issues.md]: not found

File: docs/model/introduction-to-model-architecture.md [hunk 7]
```diff
@@ -1,21 +1,21 @@
-# Basics About Model Architecture
+# Introduction to Model Architecture
```
Comment (boris-brtan) [inline docs/model/introduction-to-model-architecture.md]: here it is :+1: found.

File: docs/model/introduction-to-model-architecture.md [hunk 8]
```diff
@@ -1,21 +1,21 @@
...
 
-In this article you will learn about [entities](#entity), [facades](#facade), [repositories](#repository) and their mutual relations.
+In this article you will learn about model, its dependencies, [entities](#entity), [facades](#facade), [repositories](#repository) and their mutual relations.
 
-## Basics about the model
+## Definition of a Model
+Definition of model is adopted form DDD definition of a model. Model is a system of abstractions that describes selected aspect of a domain.
```
Comment (boris-brtan) [inline docs/model/introduction-to-model-architecture.md]: `a model`
adopted form DDD ???? `adopted from DDD` or `adopted form of DDD` 
DDD could be explained better with some URL

File: docs/model/introduction-to-model-architecture.md [hunk 9]
```diff
@@ -1,21 +1,21 @@
-# Basics About Model Architecture
+# Introduction to Model Architecture
 
-In this article you will learn about [entities](#entity), [facades](#facade), [repositories](#repository) and their mutual relations.
+In this article you will learn about model, its dependencies, [entities](#entity), [facades](#facade), [repositories](#repository) and their mutual relations.
 
-## Basics about the model
+## Definition of a Model
```
Comment (boris-brtan) [inline docs/model/introduction-to-model-architecture.md]: 2ndLvl titles are not Capitalised
```suggestion
## Definition of a model
```

File: docs/model/introduction-to-model-architecture.md [hunk 10]
```diff
@@ -1,21 +1,21 @@
...
+Definition of model is adopted form DDD definition of a model. Model is a system of abstractions that describes selected aspect of a domain.
 
-Model architecture of Shopsys Framework is inspired by Domain Driven Design (DDD). Let us define some terms first:
-- **Domain** in DDD is a sphere of knowledge or activity we build application logic around. The domain of Shopsys Framework is e-commerce.
-- **Domain model** is a system of abstractions that describes selected aspects of the domain.
-- **Domain logic** or **business logic** is the higher level rules for how objects of the domain model interact with one another.
+Domain is a sphere of knowledge or activity we build application logic around. The domain of Shopsys Framework is e-commerce.
+
+Each domain has its logic which is the higher level rules for how objects of the domain model interact with one another.
```
Comment (boris-brtan) [inline docs/model/introduction-to-model-architecture.md]: logic is the higher level rules ? maybe some commas could help with this sentence

File: docs/model/introduction-to-model-architecture.md [hunk 11]
```diff
@@ -123,7 +123,7 @@ class CartRepository
             'DELETE FROM cart_items WHERE cart_id IN (
                 SELECT C.id
                 FROM carts C
-                WHERE C.modified_at <= :timeLimit AND user_id IS NULL)',
+                WHERE C.modified_at <= :timeLimit AND user_id IS NOT NULL)',
```
Comment (boris-brtan) [inline docs/model/introduction-to-model-architecture.md]: are you sure with this?
```
    /**
     * @var \Shopsys\FrameworkBundle\Model\Customer\User|null
     *
     * @ORM\ManyToOne(targetEntity="Shopsys\FrameworkBundle\Model\Customer\User")
     * @ORM\JoinColumn(name="user_id", referencedColumnName="id", nullable = true, onDelete="CASCADE")
     */
    protected $user;
```

File: docs/model/introduction-to-model-architecture.md [hunk 12]
```diff
@@ -132,7 +132,7 @@ class CartRepository
         ]);
 
         $nativeQuery = $this->em->createNativeQuery(
-            'DELETE FROM carts WHERE modified_at <= :timeLimit AND user_id IS NULL',
+            'DELETE FROM carts WHERE modified_at <= :timeLimit AND user_id IS NOT NULL',
```
Comment (boris-brtan) [inline docs/model/introduction-to-model-architecture.md]: are you sure with this, since user_id is nullable?

File: docs/cookbook/basic-data-import.md [hunk 13]
```diff
@@ -45,7 +45,7 @@ The source is mocked with [apiari.io service](http://docs.ssfwbasicdataimportdem
 ]
 ```
 
-### Step 1 - Add `$extId` to [Product](../../project-base/src/Shopsys/ShopBundle/Model/Product/Product.php) [entity](../introduction/basics-about-model-architecture.md#entity)
+### Step 1 - Add `$extId` to [Product](../../project-base/src/Shopsys/ShopBundle/Model/Product/Product.php) [entity](../model/basics-about-model-architecture.md#entity)
```
Comment (boris-brtan) [inline docs/cookbook/basic-data-import.md]: does this exist?

File: docs/index.md (hunk 1 above)
Comment (boris-brtan) [reply docs/index.md]: also titles

File: docs/model/rules-to-model.md [hunk 14]
```diff
@@ -0,0 +1,50 @@
+# Rules to a Model
+In this article you will learn about model rules, what is and what is not model, and how to correctly create and insert new functionality into existing directory architecture.
+
+## Rules
+- classes always must be in right FQN, if you want to add a new functionality that works with product, your class should be created into `\Shopsys\ShopBundle\Model\Product`
+- classes in model must always represent a some kind of behavior of a model unit
+- model unit should always be represented by an Entity
+
+## What is and What is Not a Model
```
Comment (boris-brtan) [inline docs/model/rules-to-model.md]: 2.lvl titles does not contain Capitals

File: docs/model/introduction-to-model-architecture.md (hunk 10 above)
Comment (MattCzerner) [reply docs/model/introduction-to-model-architecture.md]: If you see some specific mistake, please point that out, because in my opinion, this sentence makes sense

File: docs/model/introduction-to-model-architecture.md (hunk 12 above)
Comment (MattCzerner) [reply docs/model/introduction-to-model-architecture.md]: Ye, sorry i copied the `deleteOldCartsForRegisteredCustomers()` instead. good point

File: project-base/src/Shopsys/ShopBundle/DataFixtures/resources/.~lock.demo-data-products.csv# [hunk 15]
```diff
@@ -0,0 +1 @@
+,mathew,mathew-Inspiron-15-5578,07.03.2019 15:26,file:///home/mathew/.config/libreoffice/4;
```
Comment (boris-brtan) [inline project-base/src/Shopsys/ShopBundle/DataFixtures/resources/.~lock.demo-data-products.csv#]: please remove this file from fixup

File: docs/model/rules-to-model.md (hunk 14 above)
Comment (boris-brtan) [reply docs/model/rules-to-model.md]: hmm i cant see any change also there are other 2.lvl titles capitalized

File: docs/model/rules-to-model.md [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+Some models does not have a persisted entity that represents a model unit. For example one of them is Feed, even though it does not have a entity, it is related to a e-commerce domain and because of that we keep it in a model directory.
+
+Units of model without persisted entity:
+* Feed
+* Newsletter
+* Heureka
+* ShopInfo
+
+## Model Grids
+<!-- TODO: add a link to a grid documentation -->
```
Comment (boris-brtan) [inline docs/model/rules-to-model.md]: we can't forget about this, but if this should be part of the releaser it needs to have one more dash `private const TODO_PLACEHOLDER = '<!--- TODO';`

**Comment by MattCzerner** [comment]
Dunno how to `how to extend model from repository and how from modules (urls to modules readmes)`, I dont know which url was meant by extensibility for modules.

File: docs/model/rules-to-model.md [hunk 17]
```diff
@@ -42,7 +47,7 @@ Units of model without persisted entity:
 * Heureka
 * ShopInfo
 
-## Model grids
+### Model grids
```
Comment (boris-brtan) [inline docs/model/rules-to-model.md]: are you sure that `## MultidomainEnityClassProvider` is outside this group ?

Comment (MattCzerner) [reply docs/model/rules-to-model.md]: yeah, these are exceptions to the model rules. MultidomainEntityClassProvider is in my opinion different story.

File: docs/index.md [hunk 18]
```diff
@@ -44,6 +43,10 @@
 * [Creating a Multidomain Design](./cookbook/creating-a-multidomain-design.md)
 * [Working with Multiple Cron Instances](./cookbook/working-with-multiple-cron-instances.md)
 
+## Model
```
Comment (LukasHeinz) [inline docs/index.md]: I think more articles should be moved to this sections - especially those regarding Entities because those are part of model. Or Not?

File: docs/model/introduction-to-model-architecture.md [hunk 19]
```diff
@@ -1,21 +1,21 @@
-# Basics About Model Architecture
+# Introduction to Model Architecture
 
-In this article you will learn about [entities](#entity), [facades](#facade), [repositories](#repository) and their mutual relations.
+In this article you will learn about model, its dependencies, [entities](#entity), [facades](#facade), [repositories](#repository) and their mutual relations.
 
-## Basics about the model
+## Definition of a model
+Definition of model is adopted from [DDD](https://stackoverflow.com/questions/1222392/can-someone-explain-domain-driven-design-ddd-in-plain-english-please/1222488#1222488) definition of a model. Model is a system of abstractions that describes selected aspect of a domain.
```
Comment (LukasHeinz) [inline docs/model/introduction-to-model-architecture.md]: I would recommend to add some info, that in project we are also using the term "domain" for something else. For example in Rules of model at the end there is "MultidomainEnityClassProvider" and it is about different "domain".

File: docs/model/rules-to-model.md [hunk 20]
```diff
@@ -0,0 +1,55 @@
...
+
+## Exceptions to rules
+Some concepts in our current model does not follow rules listed above.
+
+### Models that will be moved
+* AdminNavigation - will be moved to a components
+* AdvancedSearch - will be moved to a components
+* Breadcrumb - will be moved to a components
+* ContactForm - will be moved to a components
+* Cookies - will be moved to a components
```
Comment (PetrHeinz) [inline docs/model/rules-to-model.md]: Cookies a LegalConditions závisí na Article

File: docs/model/rules-to-model.md [hunk 21]
```diff
@@ -0,0 +1,55 @@
...
+
+Units of model without persisted entity:
+* Feed
+* Newsletter
+* Heureka
+* ShopInfo
+
+### Model grids
+<!--- TODO: add a link to a grid documentation -->
+Grids are usable in any different domain, but are located in a `\Shopsys\FrameworkBundle\<MODEL>\Grid`. These classes will be moved to a different folder.
```
Comment (PetrHeinz) [inline docs/model/rules-to-model.md]: Why it's not just listed in Models that will be moved?

File: docs/model/rules-to-model.md [hunk 22]
```diff
@@ -0,0 +1,55 @@
...
+
+## MultidomainEnityClassProvider
```
Comment (PetrHeinz) [inline docs/model/rules-to-model.md]:
```suggestion
## MultidomainEntityClassProvider
```

File: docs/model/rules-to-model.md [hunk 23]
```diff
@@ -0,0 +1,55 @@
+# Rules to a model
```
Comment (PetrHeinz) [inline docs/model/rules-to-model.md]: I would rather name it "Model Rules"
```suggestion
# Model Rules
```

Comment (PetrHeinz) [inline docs/model/rules-to-model.md]: I'm missing here the rule that Model can depend on a Component, but not vice versa.

File: docs/model/introduction-to-model-architecture.md [hunk 24]
```diff
@@ -1,21 +1,21 @@
...
 
 Domain model of Shopsys Framework is located in [`FrameworkBundle/Model`](https://github.com/shopsys/framework/tree/master/src/Model). Its concept is to separate behavior and properties of objects from its persistence. This separation is suitable for code reusability, easier testing and it fulfills the Single Responsibility Principle.
 
 Code belonging to the same feature is grouped together (eg. `Cart` and `CartItem`). Names of classes and methods are based on real world vocabulary to be more intuitive (eg. `OrderHashGenerator` or `getSellableProductsInCategory()`).
 
-Model is divided into three parts: Entity, Repository and Facade. There is `EntityManager` to access the database.
+Model is mostly divided into three parts: Entity, Repository and Facade.
 
-![model architecture schema](img/model-architecture.png 'model architecture schema')
+![model architecture schema](../introduction/img/model-architecture.png 'model architecture schema')
```
Comment (PetrHeinz) [inline docs/model/introduction-to-model-architecture.md]: The image should have been moved as well imo

File: docs/model/introduction-to-model-architecture.md [hunk 25]
```diff
@@ -193,6 +195,14 @@ class CartFacade
...
 ## Cooperation of layers
-The controller handles the request (eg. [saved data](entities.md#entity-data) from form) and passes data to the facade.
+The controller handles the request (eg. [saved data](../introduction/entities.md#entity-data) from form) and passes data to the facade.
 The facade receives data from the controller and requests appropriate entities from the repository.
 Entities and supporting classes (like recalculators, schedulers) processes data and returns output to the facade, that persist it by entity manager.
+
+## Model extension
+Entity extension is described in [Entity Extension article](../extensibility/entity-extension.md).
+
+Other parts of a model are extended by standard inheritance.
```
Comment (PetrHeinz) [inline docs/model/introduction-to-model-architecture.md]: I'm not sure whether class inheritance can be described as standard for extension. Maybe we could add a bit of an explanation:

Other parts of a model can be extended by class inheritance and adding an alias to your `services.yml`, eg.:
```yaml
services:
    Shopsys\FrameworkBundle\Model\Article\ArticleFacade: '@Shopsys\ShopBundle\Model\Article\ArticleFacade'
```

File: docs/model/model-rules.md [hunk 26]
```diff
@@ -0,0 +1,53 @@
+# Model Rules
+In this article you will learn about model rules, what is and what is not model, and how to correctly create and insert new functionality into existing directory architecture.
+
+## Hierarchy nesting
+All classes related to a model unit are nested into `\Shopsys\FrameworkBundle\Model\<MODEL>`.
+
+They can also be nested into deeper directory such as `Shopsys\FrameworkBundle\Model\Product\Search` if it encapsulate group of classes representing some specific functionality, for example search functionality for a Product entity.
+
+## Rules
+- classes always must be in right FQN, if you want to add a new functionality that works with product, your class should be created into `\Shopsys\ShopBundle\Model\Product`
```
[the remark this answers is not in the document]
Comment (MattCzerner) [reply docs/model/model-rules.md]: To always have it aligned correctly to the folders? I described it right after, if you work with Product, you should have it placed `\Shopsys\ShopBundle\Model\Product`

File: docs/model/model-rules.md [hunk 27]
```diff
@@ -0,0 +1,53 @@
...
+- classes in model must always represent a some kind of behavior of a model unit
```
[the remark this answers is not in the document]
Comment (MattCzerner) [reply docs/model/model-rules.md]: I dont understand what you mean. Repository is a basic concept of a model.

TopCategory represents the behavior of a top **category**

File: docs/model/custom-entities.md [hunk 28]
```diff
@@ -4,7 +4,7 @@ When you need to add functionality that is not in the system, like an advertisin
 then you need your own, custom entities.
 
 * The system is prepared and configured for custom entities.
-The configuration is placed in [doctrine.yml](/project-base/app/config/packages/doctrine.yml), section `doctrine.orm.mappings`.
+The configuration is placed in [doctrine.yml](/config/packages/doctrine.yml), section `doctrine.orm.mappings`.
```
Comment (PetrHeinz) [inline docs/model/custom-entities.md]: Why did you change this path? It 404's now.

File: docs/model/how-to-work-with-money.md [hunk 29]
```diff
@@ -40,12 +40,12 @@ The scale has to be specified during [rounding](#rounding), [creating from float
 
 ## Money Class
 
-[`Money`](/packages/framework/src/Component/Money/Money.php) is an immutable [value object](https://codete.com/blog/value-objects/).
+[`Money`](/Component/Money/Money.php) is an immutable [value object](https://codete.com/blog/value-objects/).
```
Comment (PetrHeinz) [inline docs/model/how-to-work-with-money.md]: All path changes in this file in d615476a4572b00726a31bcfea38c308548c19d0 broke the links.

File: docs/model/model-rules.md (hunk 27 above)
[the remark this answers is not in the document]
Comment (PetrHeinz) [reply docs/model/model-rules.md]: This is the whole class without comments:
```php
class TopCategory
{
    protected $category;
    protected $domainId;
    protected $position;

    public function __construct(Category $category, $domainId, $position)
    {
        $this->category = $category;
        $this->domainId = $domainId;
        $this->position = $position;
    }

    public function getCategory()
    {
        return $this->category;
    }
}
```
I don't know what you mean by the rule *"classes in model must always represent a some kind of behavior of a model unit"*. Can you please explain what you mean by that rule in relation to this class (and ideally to both other classes)?

File: docs/model/model-rules.md (hunk 26 above)
[the remark this answers is not in the document]
Comment (PetrHeinz) [reply docs/model/model-rules.md]: Can you rewrite it to "classes always must be in the right namespace" then?

File: docs/model/introduction-to-model-architecture.md [hunk 30]
```diff
@@ -1,21 +1,21 @@
...
 
-## Basics about the model
+## Definition of a model
+Definition of model is adopted from [DDD](https://stackoverflow.com/questions/1222392/can-someone-explain-domain-driven-design-ddd-in-plain-english-please/1222488#1222488) definition of a model. Model is a system of abstractions that describes selected aspect of a domain.
 
-Model architecture of Shopsys Framework is inspired by Domain Driven Design (DDD). Let us define some terms first:
-- **Domain** in DDD is a sphere of knowledge or activity we build application logic around. The domain of Shopsys Framework is e-commerce.
-- **Domain model** is a system of abstractions that describes selected aspects of the domain.
-- **Domain logic** or **business logic** is the higher level rules for how objects of the domain model interact with one another.
+Domain is a sphere of knowledge or activity we build application logic around. The domain of Shopsys Framework is e-commerce. In Shopsys Framework we also use term domain for another use case, which is the domain url address and things related to them, you can read more about Domain in [Domain, Multidomain, Multilanguage](../introduction/domain-multidomain-multilanguage.md)
```
[the remark this answers is not in the document]
Comment (PetrHeinz) [reply docs/model/introduction-to-model-architecture.md]: The full copy of the paragraph seems a bit too much. It takes the focus away from domain model. Why did you use it instead of my suggestion? Do you think it's important to explain it fully here?

File: docs/model/introduction-to-model-architecture.md [hunk 31]
```diff
@@ -193,6 +204,21 @@ class CartFacade
...
+## Model extension
+Entity extension is described in [Entity Extension article](../extensibility/entity-extension.md).
+
+Other parts of a model can be extended by class inheritance and adding an alias to your `services.yml`, eg.:
+```
+services:
+    Shopsys\FrameworkBundle\Model\Article\ArticleFacade: '@Shopsys\ShopBundle\Model\Article\ArticleFacade'
+```
+
+### Extending from bundle
```
Comment (PetrHeinz) [inline docs/model/introduction-to-model-architecture.md]:
```suggestion
### Extending from a bundle
```

File: docs/index.md [hunk 32]
```diff
@@ -44,6 +38,15 @@
...
+* [Introduction to Model Architecture](./model/introduction-to-model-architecture.md)
+* [Rules to Model](./model/model-rules.md)
```
[the remark this answers is not in the document]
Comment (PetrHeinz) [reply docs/index.md]: It seems that you forgot it again and marked the comment as resolved.

File: docs/model/model-rules.md [hunk 33]
```diff
@@ -0,0 +1,53 @@
...
+* Cookies - will be moved under a Article model since it is closely related to a Article
+* Grids - Grids are located in a `\Shopsys\FrameworkBundle\<MODEL>\Grid` namespace, these will be moved to a separate Grid directory
+* Mail - will be moved to a components
+* Module - will be moved to a components
+* Sitemap - will be moved to a components
+* Localization - will be moved to a components
+* LegalConditions - will be moved under a Article model since it is closely related to a Article
+* Seo- will be moved to a components
+* Slider - will be moved to a components
+* Statistics - will be moved to a components
```
[the remark this answers is not in the document]
Comment (PetrHeinz) [reply docs/model/model-rules.md]: Please, fix it where it was copypasted as well.

File: docs/model/model-rules.md [hunk 34]
```diff
@@ -0,0 +1,53 @@
...
+
+They can also be nested into deeper directory such as `Shopsys\FrameworkBundle\Model\Product\Search` if it encapsulate group of classes representing some specific functionality, for example search functionality for a Product entity.
+
+## Rules
+- classes always must be in right FQN, if you want to add a new functionality that works with product, your class should be created into `\Shopsys\ShopBundle\Model\Product`
+- classes in model must always represent a some kind of behavior of a model unit
+- model unit should always be represented by an Entity
+- model can be dependent on a component but not vice versa, this rule comes from definition of a [Component](../introduction/components.md).
+
+## What is and what is not a model
```
Comment (PetrHeinz) [inline docs/model/model-rules.md]: Here I'm missing the mention of integration code not being a part of the Model namespace. Typically controllers, forms or grids (which are a part of Model atm but should be moved).

Otherwise, according to "That means that everything in a model should be related to some functionality of Domain, in our case, e-commerce." it should be a part of the Model namepsace. Eg. `ProductType` is related to the functionality of editing products.

File: docs/model/model-rules.md [hunk 35]
```diff
@@ -0,0 +1,53 @@
...
+* Seo- will be moved to a components
+* Slider - will be moved to a components
+* Statistics - will be moved to components
+
+### Model without persisted entity representation
+Some models does not have a persisted entity that represents a model unit. For example one of them is Feed, even though it does not have a entity, it is related to a e-commerce domain and because of that we keep it in a model directory.
+
+Units of model without persisted entity:
+* Feed
+* Newsletter
```
Comment (PetrHeinz) [inline docs/model/model-rules.md]: Newsletter has the `NewsletterSubscriber` entity.

File: docs/model/custom-entities.md (hunk 28 above)
Comment (MattCzerner) [reply docs/model/custom-entities.md]: fixed

File: docs/model/how-to-work-with-money.md (hunk 29 above)
Comment (MattCzerner) [reply docs/model/how-to-work-with-money.md]: fixed

File: docs/model/model-rules.md (hunk 27 above)
[the remark this answers is not in the document]
Comment (MattCzerner) [reply docs/model/model-rules.md]: Decided not to have this rule included.

**Comment by MattCzerner** [comment]
Hope I covered it all. I will squash all commits together at the end.
